### PR TITLE
feat(accounts) Only show /accounts/confirm-email/ link confirmation process to unverified email addresses

### PR DESCRIPTION
**Pull request by 9mido**
Attempts to solve: 

https://github.com/pennersr/django-allauth/issues/1522

https://github.com/pennersr/django-allauth/issues/2862

Currently when a user signs up, they get an email with a link where they are redirected to /accounts/confirm-email/ to confirm their account is legitimate. After clicking their /accounts/confirm-email/ link confirm button, they can re-use that same /accounts/confirm-email/ link again in the same browser or a different browser and go through the process of confirming again. 

With this new setting turned set to True, a django developer could decide to not allow verified email addresses to confirm themselves again with the same link after already verifying themselves the first time. It will instead show the message "This e-mail confirmation link expired or is invalid" the second time around. Otherwise, with this setting set to False, the email address could be verified again with the same link after they have already verified themselves and need to wait until ACCOUNT_EMAIL_CONFIRMATION_EXPIRE_DAYS runs out to no longer confirm themselves which is the current allauth behavior.

### CODE REVIEW

File: allauth/account/views.py [hunk 1]
```diff
@@ -293,6 +293,9 @@ def get(self, *args, **kwargs):
             self.object = self.get_object()
             if app_settings.CONFIRM_EMAIL_ON_GET:
                 return self.post(*args, **kwargs)
+            email = self.object.email_address.email
+            if EmailAddress.objects.filter(email=email, verified=True).exists() and app_settings.EMAIL_CONFIRMATION_UNVERIFIED_ONLY is True:
```
Comment (pennersr) [inline allauth/account/views.py:297]: Doing this check here results in this not working when `CONFIRM_EMAIL_ON_GET` is `True`.

Wouldn't `EmailConfirmationHMAC::from_key()` be a better place? The `EmailAddress` lookup is already there.

File: allauth/account/app_settings.py [hunk 2]
```diff
@@ -315,6 +315,10 @@ def SALT(self):
     def PRESERVE_USERNAME_CASING(self):
         return self._setting("PRESERVE_USERNAME_CASING", True)
 
+    @property
+    def EMAIL_CONFIRMATION_UNVERIFIED_ONLY(self):
+        return self._setting("EMAIL_CONFIRMATION_UNVERIFIED_ONLY", True)
```
Comment (pennersr) [inline allauth/account/app_settings.py:320]: Wondering why would this need to be a setting, as having a setting means that there are good reasons to turn this switch either way.  Is there a good reason why `True` should not be the only/default behaviour ?

**Review by pennersr — CHANGES_REQUESTED** [review]

File: allauth/account/app_settings.py (hunk 2 above)
Comment (9mido) [reply allauth/account/app_settings.py:320]: You're right. I don't think there needs to be a setting here and it should be True all the time. This setting can be removed then.

File: allauth/account/views.py (hunk 1 above)
Comment (9mido) [reply allauth/account/views.py:297]: Ok good catch I am fine with that.

File: allauth/account/views.py [hunk 3]
```diff
@@ -293,6 +293,7 @@ def get(self, *args, **kwargs):
             self.object = self.get_object()
             if app_settings.CONFIRM_EMAIL_ON_GET:
                 return self.post(*args, **kwargs)
+            email = self.object.email_address.email
```
Comment (pennersr) [inline allauth/account/views.py:296]: This is not used I think?

File: allauth/account/app_settings.py [hunk 4]
```diff
@@ -314,7 +314,7 @@ def SALT(self):
     @property
     def PRESERVE_USERNAME_CASING(self):
         return self._setting("PRESERVE_USERNAME_CASING", True)
-
+
```
Comment (pennersr) [inline allauth/account/app_settings.py:317]: Wrongly formatted...

**Pull request merged**